crepecake/src/test/java/com/google/cloud/tools/crepecake/json/JsonHelperTest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.json;
+
+import com.google.cloud.tools.crepecake.image.DescriptorDigest;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.security.DigestException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.gradle.internal.impldep.com.google.common.io.CharStreams;
+import org.junit.Assert;
+import org.junit.Test;
+
+/** Tests for {@link JsonHelper}. */
+public class JsonHelperTest {
+
+  private static class TestJson extends JsonTemplate {
+    private int number;
+    private String text;
+    private DescriptorDigest digest;
+    private InnerObject innerObject;
+    private List<InnerObject> list;
+
+    private static class InnerObject extends JsonTemplate {
+      private int number;
```

Comment:
this field shadows the above field. Rename to avoid confusion.

Reply:
I named this the same intentionally to check an inner JSON object with a field having the same name works fine.

Reply:
In that case, add a comment making this explicit.